os/drivers/lwnl/lwnl_evt_queue.c

```diff
@@ -22,24 +22,21 @@
 #include <errno.h>
 #include <debug.h>
 #include <net/if.h>
+#include <pthread.h>
 #include <tinyara/net/if/wifi.h>
 #include "lwnl_evt_queue.h"
+#include "lwnl_log.h"
 
-#define LWQ_LOCK
-#define LWQ_UNLOCK
+static sem_t g_wm_sem;
```

Comment:
locally many semaphores, can you please check do we need it really? Cant we make it as part of lwnl_event  or lwnl_queue struct instead? Also semaphore with same name is used in other file, anyway static. But confusing.

Reply:
yes those are necessary. g_wm_sem is to send signal to main task and g_queue_sem is to protect event queue.
And I don't understand how to use lwnl_event in here. only lwnl can access lwnl queue.
I changed the name of those semaphores. thanks